src/metax_api/tests/api/rest/base/views/files/write.py

```diff
@@ -675,6 +676,27 @@ def test_update_file_not_found(self):
         response = self.client.put('/rest/files/doesnotexist', self.test_new_data, format="json")
         self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
 
+    def test_update_file_allowed_projects_ok(self):
+        f = self.client.get('/rest/files/1').data
+        id = f['identifier']
+        pid = f['project_identifier']
```

Comment:
Careful when calling an identifier field variable id, since there is also a field called id (i.e. the db primary key). In this case its probably ok.

project_identifier are also pretty much never referred to as "pid", better to call such variables just "project", "proj", "project_identifier", as done elsewhere in the codebase. "pid" is usually short for "persistent identifier", and in the context of files would normally refer to "identifier" field. With datasets, pids generally refer to the "preferred_identifier" field.

Although since the tests are so short, may be a good idea to simply use f['identifier'] and project field where needed, instead of using intermediary variables.